frontend/client/modules/create/sagas.ts

```diff
@@ -64,6 +73,34 @@ export function* handleSaveDraft(): SagaIterator {
   }
 }
 
+export function* handleFetchAndCreateDrafts(
+  action: ReturnType<typeof fetchAndCreateDrafts>,
+): SagaIterator {
+  yield put(fetchDrafts());
+  yield take([types.FETCH_DRAFTS_FULFILLED, types.FETCH_DRAFTS_PENDING]);
+  const drafts: Yielded<typeof getDrafts> = yield select(getDrafts);
+
+  // Back out if draft fetch failed and we don't have drafts
+  if (!drafts) {
+    console.warn('Fetch of drafts failed, not creating new draft');
+    return;
+  }
+
+  // Create new draft if we don't have one for rfp (if provided) or if
+  // we don't have any at all
+  if (action.payload.rfpId) {
+    const rfpDraft = drafts.find(d => !!d.rfp && d.rfp.id === action.payload.rfpId);
+    if (!rfpDraft) {
+      yield put(createDraft(action.payload));
+    } else if (action.payload.redirect) {
+      // If we were supposed to redirect, redirect to existing rfp draft
+      yield put(push(`/proposals/${rfpDraft.proposalId}/edit`));
```

Comment:
No biggie, but could do a `replace` instead of a `push` to keep it out of the history. This goes for many of our pushes, and <Redirect.../> s.

Just sth to think about.